src/encoding_implementations/amplitude_encoding.jl

```diff
@@ -0,0 +1,68 @@
+export AmplitudeEncoding
+
+"""
+    AmplitudeEncoding <: Encoding
```

Comment:
So to get things straight, the `AmplitudeAwareOrdinalPattern` outcome space is the _combination_ encoding of `OrdinalPatternEncoding` and `AmplitudeEncoding`? 

Unfortunately the name `AmplitudeEncoding` is too simple, while this encoding is really complex, and its name doesn't convey what it does. My mind went into value histogram with the name.

More importantly, is this encoding useful in the first place? It hasn't been used in the literature as far as I understand. We should be mindful into not adding too many things in the code base that may have no use on their own.

Reply:
As far as I can tell, this PR does not alter any of the `probabilities` function calls. So, I have to wonder, should we first do some research that shows that all these extra functionality is useful in some way?

Reply:
I just don't see the reason for existence of an encoding without a corresponding outcome space to use it.

Reply:
> So to get things straight, the AmplitudeAwareOrdinalPattern outcome space is the combination encoding of OrdinalPatternEncoding and AmplitudeEncoding?

When used with `probabilities`, `AmplitudeAwareOrdinalPattern` is (roughly) the combination of `OrdinalPatternEncoding` with a linear combination of `AmplitudeEncoding` and `FirstDifferenceEncoding` dictating the weights. 

> I just don't see the reason for existence of an encoding without a corresponding outcome space to use it.

As stated in the bottom statement in the initial PR comment, the reason is to have complete flexibility in how to discretize data in upstream functions. I am using the new encodings in an upcoming paper to develop some information transfer algorithms that improve on the "naive"/only ordinal pattern based ones. 

They are not yet in any papers, because nobody has used them in this way yet.